### PR TITLE
UX-610 Convert ActionList to typescript

### PR DESCRIPTION
**Pull request by jonambas**
Note: `ts-loader` is extremely slow

### What Changed
- Converts `ActionList` and all its components to typescript
- Fixes `IconBase` types

### How To Test or Verify
- Verify the build works and type definitions for `ActionList` are generated
- Verify libby and the libby build work

### PR Checklist

- [ ] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/ActionList/ActionList.tsx [hunk 1]
```diff
@@ -86,7 +108,7 @@ const ActionList = React.forwardRef(function ActionList(props, userRef) {
   function assignRefs(node) {
     wrapperRef.current = node;
     if (userRef) {
-      userRef.current = node;
+      (userRef as React.MutableRefObject<HTMLDivElement>).current = node;
```
Comment (jonambas) [inline packages/matchbox/src/components/ActionList/ActionList.tsx:111]: refs are immutable, this overrides it

File: libby/action/ActionList.lib.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { describe, add } from '@sparkpost/libby-react';
-import { ActionList, Button, Inline, Popover, Panel, Box } from '@sparkpost/matchbox';
+import { Button, Inline, Popover, Panel, Box } from '@sparkpost/matchbox';
+import { ActionList } from '../../packages/matchbox/src';
```
Comment (jonambas) [inline libby/action/ActionList.lib.tsx:4]: I couldn't figure out how to get types imported through the webpack alias. might need to convert all stories to this import structure to get type checking

Comment (logansparlin) [reply libby/action/ActionList.lib.tsx:5]: I'm pretty sure you have to set the webpack alias and then set a matching url in a `paths` object in tsconfig.json like what we do here https://github.com/SparkPost/design.sparkpost.com/blob/main/tsconfig.json#L17

File: packages/matchbox/src/components/ActionList/Action.tsx [hunk 3]
```diff
@@ -6,14 +6,56 @@ import { Box } from '../Box';
...
+  /**
+   * @deprecated Compose actions with ActionList.Action instead
+   */
+  visible?: boolean;
+};
+
+type LinkProps = {
+  is?: 'link';
+  to?: string;
+  external?: boolean;
```
[the remark this answers is not in the document]
Comment (logansparlin) [reply packages/matchbox/src/components/ActionList/Action.tsx:39]: Probably a good idea to abstract since we'll be reusing

**Review by logansparlin — APPROVED** [review]
This looks good, just a couple comments

**Comment by logansparlin** [comment]
Could merge as is, but would be nice to get aliases working and abstract out reusable types

**Pull request merged**